File: reverse-integer/SolutionStringReverse.java
class SolutionStringReverse {
    public int reverse(int x) {
        
        if(x == 0) return 0;

        StringBuffer absX = new StringBuffer(String.valueOf(java.lang.Math.abs(x)));
        String outputStr;
        if(x > 0){
            outputStr = absX.reverse().toString();
        } else {
            outputStr = "-" + absX.reverse().toString();
        }

        try{
            return Integer.parseInt(outputStr);
        } catch(Exception e){
            return 0;
        }
    }
}

File: reverse-integer/SolutionStringReverse.php
<?php
class SolutionStringReverse {

    /**
     * @param Integer $x
     * @return Integer
     */
    function reverse($x) {
        
        if($x === 0) return 0;

        $absX = abs($x);

        if($x < 0){
            $output = "-".strrev(strval($absX));
        } else {
            $output = strrev(strval($absX));
        }

        if($output >= -2147483648 && $output <= 2147483647) {
            return intval($output);
        } else {
            return 0;
        }

    }
}